src/aggregation/staging_collector.rs

```diff
@@ -0,0 +1,83 @@
+use super::agg_req_with_accessor::AggregationsWithAccessor;
+use super::intermediate_agg_result::IntermediateAggregationResults;
+use super::segment_agg_result::SegmentAggregationCollector;
+use crate::DocId;
+
+pub(crate) const DOC_BLOCK_SIZE: usize = 64;
+pub(crate) type DocBlock = [DocId; DOC_BLOCK_SIZE];
+
+/// The StagingAggregationCollector is the generic version of the collector, which
+/// can handle arbitrary complexity of  sub-aggregations. Ideally we never have to pick this one
+/// and can provide specialized versions instead, that remove some of its overhead.
+#[derive(Clone)]
+pub(crate) struct StagingAggregationCollector<T> {
```

Comment:
That's just a suggestion, but isn't it the definition of Buffering?

```suggestion
pub(crate) struct BufAggregationCollector<T> {
```